### PR TITLE
Add com.kjxbyz.PicGuard

### PR DESCRIPTION
**Pull request by kjxbyz**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
  
Introducing PicGuard - the ultimate solution for securing and customizing your digital photos. With our app, you can easily add watermarks to your images to protect your creative work and maintain your privacy.</li>

PicGuard offers robust privacy protection features, ensuring your personal and sensitive information remains safeguarded. It's highly customizable, allowing you to personalize watermarks with various fonts, styles, and designs to suit your needs.

Accessible on web, mobile, and desktop, you can seamlessly integrate and manage your watermarking tasks across all your devices. PicGuard, as an open-source tool, it's not only free but also continuously improved by a community of developers dedicated to providing the best user experience.

Download PicGuard today and take control of your photo security and personalization like never before!

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: com.kjxbyz.PicGuard.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+id: com.kjxbyz.PicGuard
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
```
Comment (bbhtt) [inline com.kjxbyz.PicGuard.yml:4]: Does this actually need gnome runtime? Or freedesktop will work too?

**Comment by bbhtt** [comment]
bot, build com.kjxbyz.PicGuard

**Comment by bbhtt** [comment]
Version key in desktop file is not your app version, it is the spec version. https://specifications.freedesktop.org/desktop-entry-spec/latest/recognized-keys.html

File: com.kjxbyz.PicGuard.yml (hunk 1 above)
Comment (kjxbyz) [reply com.kjxbyz.PicGuard.yml:4]: I developed it using flutter, and I'm not sure if I have to use the `gnome` runtime, but I think it should work fine on `freedesktop`. How can I support both runtimes at the same time?

**Comment by kjxbyz** [comment]
> Version key in desktop file is not your app version, it is the spec version. https://specifications.freedesktop.org/desktop-entry-spec/latest/recognized-keys.html

So how do I specify the version of my application in the manifest?

**Comment by bbhtt** [comment]
In the `release` tag of metainfo https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#release

File: com.kjxbyz.PicGuard.yml (hunk 1 above)
Comment (bbhtt) [reply com.kjxbyz.PicGuard.yml:4]: Can you change it to freedesktop? if it builds and runs, then it is fine to use that.

GNOME is a superset of the freedesktop runtime, you don't need to support both.

Comment (kjxbyz) [reply com.kjxbyz.PicGuard.yml:4]: Then I need to change to freedesktop and test it locally.

File: com.kjxbyz.PicGuard.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      - install -Dm644 usr/share/icons/hicolor/128x128/apps/picguard.png /app/share/icons/hicolor/128x128/apps/com.kjxbyz.PicGuard.png
+      - install -Dm644 usr/share/icons/hicolor/256x256/apps/picguard.png /app/share/icons/hicolor/256x256/apps/com.kjxbyz.PicGuard.png
+      - install -Dm644 com.kjxbyz.PicGuard.appdata.xml /app/share/appdata/com.kjxbyz.PicGuard.appdata.xml
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://github.com/picguard/picguard/releases/download/v1.0.0+391/picguard_1.0.0+391_linux_amd64.deb
+        dest-filename: picguard.deb
+        sha256: bbc0f23ee5fde6d29c61c73063ee4c5d0e54bfff7c24ebe13d2874eaeed57553
```
[the remark this answers is not in the document]
Comment (kjxbyz) [reply com.kjxbyz.PicGuard.yml]: Because there is a `deb` package that has been generated

**Comment by kjxbyz** [comment]
Updated

File: com.kjxbyz.PicGuard.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+id: com.kjxbyz.PicGuard
+
+# https://docs.flatpak.org/en/latest/available-runtimes.html
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (bbhtt) [inline com.kjxbyz.PicGuard.yml:5]:
```suggestion
runtime-version: '23.08'
```

22.08 will be EOL in a month

**Comment by bbhtt** [comment]
bot, build com.kjxbyz.PicGuard

**Comment by kjxbyz** [comment]
Can you help me figure out why the build fails on ARM64? I don't have an ARM64 machine.

https://buildbot.flathub.org/#/builders/40/builds/13076/steps/7/logs/stdio

**Comment by bbhtt** [comment]
Seems a random error in CI, worked now.

File: com.kjxbyz.PicGuard.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  # Show windows with Wayland
+  - --socket=wayland
+  # GPU acceleration if needed
+  - --device=dri
+  # Access the network
+  - --share=network
+  # Access the XDG download directory
+  - --filesystem=xdg-download
+  # Access the XDG pictures directory
+  - --filesystem=xdg-pictures
```
Comment (hfiguiere) [inline com.kjxbyz.PicGuard.yml:26]:
```suggestion
  - --share=ipc
  - --socket=fallback-x11
  - --socket=wayland
  - --device=dri
  - --share=network
  - --filesystem=xdg-download
  - --filesystem=xdg-pictures
```

Comment (kjxbyz) [reply com.kjxbyz.PicGuard.yml]: Resolved

**Comment by kjxbyz** [comment]
Updated

**Comment by bbhtt** [comment]
bot, build com.kjxbyz.PicGuard

**Pull request closed without merging**